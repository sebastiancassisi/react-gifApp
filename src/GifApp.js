import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifApp = ({ defaultCategories = [] }) => {
  const [categories, setCategory] = useState(defaultCategories);

  return (
    <>
      <h2>GifApp</h2>
      <AddCategory setCategory={setCategory} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
