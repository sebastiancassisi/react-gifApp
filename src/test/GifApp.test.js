import { React } from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifApp } from '../GifApp';

describe('Pruebas en GifApp.js', () => {
  test('Debe mostrarse correctamente', () => {
    const wrapper = shallow(<GifApp />);
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe de mostrar una lista de categorias', () => {
    const categories = ['batman', 'ramones'];
    const wrapper = shallow(<GifApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
