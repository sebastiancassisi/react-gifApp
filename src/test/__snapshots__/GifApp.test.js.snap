// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifApp.js Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="batman"
      key="batman"
    />
    <GifGrid
      category="ramones"
      key="ramones"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifApp.js Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
